llm/src/mosaic_gpt.py

```diff
@@ -113,11 +122,15 @@ class GPTBlock(nn.Module):
 
     def __init__(self, cfg: DictConfig, device: Optional[str] = None):
         super().__init__()
+        if cfg.get('alibi', False):
+            assert 'triton' in cfg.attn_impl, 'Only triton kernel supports alibi'
         self.ln_1 = nn.LayerNorm(cfg.d_model, device=device)
         if cfg.attn_impl == 'torch':
             self.causal_attn = TorchCausalAttention(cfg, device)
         elif cfg.attn_impl == 'flash':
             self.causal_attn = FlashCausalAttention(cfg, device)
+        elif 'triton' in cfg.attn_impl:
```

Comment:
Could you elaborate why we're looking for `'triton' in cfg.attn_impl` rather than a fixed key like `cfg.attn_impl == 'triton'`?

Reply:
just me being dumb, I'll update it to follow convention.